### PR TITLE
Generic applications docs

### PR DESCRIPTION
**Pull request by f0ssel**
Todo:
- [x] Add feature flag flow steps

### CODE REVIEW

**Comment by khorne3** [comment]
Hey @f0ssel, small change: we're targeting `main` for all our new docs so that they can be published and be linked to from the product. However, we don't want them shown, so in `manifest.json` features `navigable` which turns on/off whether the doc shows in the nav bar or not. The release branch will flip all `navigable` from `false` to `true`.

I realize this is a lot, so I'll be writing this up soon

**Comment by khorne3** [comment]
[preview](https://codercom-1va4br3iu-codercom.vercel.app/docs/coder/latest/workspaces/applications)

File: workspaces/applications.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+## Application specification file
+
+The application specification file allows you to define what Coder needs to
+launch the application. Within the file, you can specify the following for each
+application:
+
+```yaml
+# /coder/apps/config.yaml
+
+apps:
```
Comment (khorne3) [inline workspaces/applications.md:23]: are any of these optional, or are they all required?

File: workspaces/applications.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    dir: /home/coder
+    # File path to icon used in application launcher
+    icon_path: /home/coder/goland.svg
+    # Command to start the application
+    command: /home/coder/.local/bin/projector
+    # Array of arguments for command
+    args: ["run"]
+    # Health checks to get running application status
+    # Can use exec or http health checks to localhost
+    health_check:
```
Comment (bpmct) [inline workspaces/applications.md:40]: does this just check for exit code 0?

File: workspaces/applications.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+application:
+
+```yaml
+# /coder/apps/config.yaml
+
+apps:
+  # Name of application in launcher
+  - name: projector
+    # Application scheme
+    scheme: http
```
Comment (bpmct) [inline workspaces/applications.md:27]: what other schemes are supported? http, ssh, mailto, or anything?

**Review by bpmct — COMMENTED** [review]
Can you also add steps to debug? I have a `/coder/apps/config.yaml` in my workspace but the icon isn't showing up.

File: workspaces/applications.md (hunk 2 above)
Comment (bpmct) [reply workspaces/applications.md:40]: what file types are supported?

File: workspaces/applications.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+The application specification file allows you to define what Coder needs to
+launch the application. Within the file, you can specify the following for each
+application:
+
+```yaml
+# /coder/apps/config.yaml
+
+apps:
+  # Name of application in launcher
+  - name: projector
```
Comment (bpmct) [inline workspaces/applications.md:25]: what can be in the name?

File: workspaces/applications.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
+---
+title: "Applications"
+description:
+  Learn how to access web applications running on your remote workspace.
+state: beta
```
Comment (vapurrmaid) [inline workspaces/applications.md:5]: should this be `alpha` ?

Comment (f0ssel) [reply workspaces/applications.md:5]: yes

Comment (scsmithr) [reply workspaces/applications.md:5]: Changed to "beta" in product instead.

File: workspaces/applications.md (hunk 1 above)
Comment (scsmithr) [reply workspaces/applications.md:23]: I believe these are all required. The only exception is one of `http` or `exec` needs to be specified.

cc @sreya to confirm.

File: workspaces/applications.md (hunk 3 above)
Comment (scsmithr) [reply workspaces/applications.md:27]: Will need to be either "http" or "https". The backend is doing a reverse proxy and uses this scheme for dialing.

File: workspaces/applications.md (hunk 2 above)
Comment (scsmithr) [reply workspaces/applications.md:40]: The `exec` health check checks for exit 0. The `http` health check checks that 200 is returned,

File: workspaces/applications.md (hunk 1 above)
Comment (f0ssel) [reply workspaces/applications.md:23]: Looking through the code, it seems @scsmithr is right, everything top level field is required and 1 health check (can optionally do both).

**Comment by bpmct** [comment]
> Can you also add steps to debug? I have a `/coder/apps/config.yaml` in my workspace but the icon isn't showing up.

This was because the frontend wasn't merged yet, but I think we should still include _some_ way to debug the generic apps, even if it's just an error in the console

File: workspaces/applications.md [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+      http:
+        scheme: "http"
+        path: "/"
+        port: 9999
+```
+
+**Notes**:
+
+- All top-level fields in the `config.yaml` file are required
+- You must include at least one health check, though we recommend both. The
```
Comment (scsmithr) [inline workspaces/applications.md:60]:
```suggestion
- You must include at least one health check. The
```

We actually only check one on the backend. If the http health check is there, we use that and only that (even if the exec health check is there).

File: workspaces/applications.md (hunk 4 above)
Comment (scsmithr) [reply workspaces/applications.md:32]: I don't believe we have any restrictions on the name. Can have space, emojis, etc.

**Review by scsmithr — APPROVED** [review]

**Pull request merged**